lib/fetcher.js

```diff
@@ -0,0 +1,30 @@
+'use strict'
+
+const co = require('co')
+
+function * addon (heroku, app, db) {
+  const {resolve} = require('heroku-cli-addons')
+  let attachment = yield resolve.attachment(heroku, app, db)
+  return yield heroku.get(`/addons/${attachment.addon.id}`)
+}
+
+function * all (heroku, app) {
+  const sortBy = require('lodash.sortby')
+  const notDatabaseUrl = (a) => !a.config_vars.find((c) => c === 'DATABASE_URL')
+
+  // TODO: make this more efficient without having to call /addons
+  // (will require an API change of some sort)
+  let {attachments, addons} = yield {
+    attachments: heroku.get(`/apps/${app}/addon-attachments`),
+    addons: heroku.get('/addons')
+  }
+  addons = addons.filter(a => a.addon_service.name === 'heroku-postgresql')
```

Comment:
Sure. There's basically three things that are different for staging add-ons:

 * the addon slug (`heroku-postgresql-maciek`)
 * the addon prefix (`HEROKU_POSTGRESQL_MACIEK_PUCE_URL`)
 * the api endpoint (`shogun-maciek.herokuapp.com`)

I *think* we've eliminated the dependency on the second one in the CLI, but the first is handled [here](https://github.com/heroku/heroku/blob/master/lib/heroku/helpers/heroku_postgresql.rb#L120) and the third [here](https://github.com/heroku/heroku/blob/master/lib/heroku/client/heroku_postgresql.rb#L26) and [here](https://github.com/heroku/heroku/blob/master/lib/heroku/client/heroku_postgresql_backups.rb#L35) (the method those two call is below). I think you can drop support for the alternate `SHOGUN` env var--that's an obsolete legacy mechanism.